lib/rubocop/cop/primer/deprecated_arguments.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "rubocop"
+require "primer/classify/utilities"
+require "primer/view_components/statuses"
+require_relative "../../../../app/lib/primer/view_helper"
+
+# :nocov:
+module RuboCop
+  module Cop
+    module Primer
+      # This cop ensures that components don't use deprecated arguments
+      #
+      # bad
+      # Component.new(foo: :deprecated)
+      #
+      # good
+      # Component.new(foo: :bar)
+      class DeprecatedArguments < RuboCop::Cop::Cop
+        INVALID_MESSAGE = <<~STR
+          Avoid using deprecated arguments: https://primer.style/view-components/deprecated.
+        STR
+
+        def on_send(node)
+          return unless valid_node?(node)
+          return unless node.arguments?
+
+          # we are looking for hash arguments and they are always last
+          kwargs = node.arguments.last
+
+          return unless kwargs.type == :hash
+
+          @deprecated_arguments = cop_config["Deprecated"] || {}
+
+          kwargs.pairs.each do |pair|
+            # Skip if we're not dealing with a symbol
+            next if pair.key.type != :sym
+            next if pair.value.type != :sym
```

Comment:
could we support strings like `color: "blue"`? Or do we never support strings for colors?